test/ruby/test_variable.rb

```diff
@@ -155,6 +155,20 @@ def test_include_refined_module_class_variable
     end
   end
 
+  class SetCvar
+    def self.set(val)
+      @@a = val # inline cache
+    end
+  end
+
+  def test_set_cvar_on_frozen_object
+    SetCvar.set(1) # fill write cache
+    SetCvar.freeze
+    assert_raise(FrozenError) do
+      SetCvar.set(2) # hit write cache, but should check frozen status
+    end
+  end
```

Comment:
```suggestion
  def test_set_cvar_on_frozen_object
    set_cvar = EnvUtil.labeled_class("SetCVar") do
      def self.set(val)
        @@a = val # inline cache
      end
    end

    set_cvar.set(1) # fill write cache
    set_cvar.freeze
    assert_raise(FrozenError) do
      set_cvar.set(2) # hit write cache, but should check frozen status
    end
  end
```

Reply:
Oh yeah that's cool, and it'll be GC'd like any old object. Thanks!